### DISCUSSION TITLE
Responsive Styles

### DISCUSSION THREAD

**Discussion by mathewgurney** [joe-bell/cva#42] [Q&A] [answered]
Hi Joe, thanks for all the hard work you've put into this package 🙏🏼  I'm coming from StitchesJs where there's the option to define responsive styles by breakpoint as seen below:

![Screenshot 2022-10-11 at 14 06 10](https://user-images.githubusercontent.com/64473170/195099835-9c2f3793-eb6d-4633-871f-31f70adc80d5.png)

Is something like this possible with the CVA API? 

Many thanks, 

Mathew

**Comment by joe-bell** [comment, marked as answer]
I'm really keen to keep `class-variance-authority` as framework-agnostic as possible, and honestly this is very tricky to achieve without knowing how the user handles breakpoints 😕 

I have some ideas for the future, but it's unlikely they'll happen anytime soon

2 approaches you can try in the meantime:

1. Hide / show different components based on the viewport
2. Create a manual responsive variant `color: 'violetFromMd'`

---

**Update**: Added to the README

https://github.com/joe-bell/cva#how-can-i-create-responsive-variants-like-stitchesjs
